spectrum_io/raw/msraw.py

```diff
@@ -120,7 +122,9 @@ def get_file_list(source: Union[str, List[str]], ext: str = "mzml"):
         return file_list
 
     @staticmethod
-    def _get_scans_pymzml(file_path: str, data: Dict, scanidx: Optional[List] = None, *args, **kwargs) -> None:
+    def _get_scans_pymzml(
+        file_path: Union[str, Path], data: Dict, scanidx: Optional[List] = None, *args, **kwargs
```

Comment:
you need to add a check with isinstance here and remove the os.path.splitext in line 138, replace this with a pathlib method and make sure the subsequent code still works or can be cleaned up